### PR TITLE
fix(core): Unsafe isAxiosError property access (no-changelog)

### PR DESCRIPTION
**Pull request by michael-radency**
## Summary

in some cases the actual error does not get returned. It is apparently reproducible with the OpenAI node but will probably also be a bug and so a problem in other nodes:

https://community.n8n.io/t/problem-in-node-openai-cannot-read-properties-of-undefined-reading-isaxioserror/34716/9

## Related tickets and issues

https://linear.app/n8n/issue/NODE-1028/actual-error-does-not-always-get-returned

## Review / Merge checklist
- [ ] PR title and summary are descriptive. **Remember, the title automatically goes into the changelog. Use `(no-changelog)` otherwise.** ([conventions](https://github.com/n8n-io/n8n/blob/master/.github/pull_request_title_conventions.md))
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included.
   > A bug is not considered fixed, unless a test is added to prevent it from happening again.
   > A feature is not complete without tests.

### CODE REVIEW

File: packages/workflow/src/RoutingNode.ts [hunk 1]
```diff
@@ -243,7 +243,7 @@ export class RoutingNode {
 					message: routingError?.message,
 					description: routingError?.description,
 					httpCode:
-						routingError.isAxiosError && routingError.response
+						routingError?.isAxiosError && routingError?.response
```
[the remark this answers is not in the document]
Comment (ivov) [reply packages/workflow/src/RoutingNode.ts:246]: > or, if this caused by line 238, then we should fix that line instead.

@michael-radency This is the case. We need to fix this at source.

**Review by ivov — COMMENTED** [review]
Do we still need all the optional access operators?

**Comment by michael-radency** [comment]
> Do we still need all the optional access operators?

@ivov 
do you think there would be downside to leaving them?

**Comment by ivov** [comment]
> > Do we still need all the optional access operators?
> 
> @ivov do you think there would be downside to leaving them?

My reasoning is that leaving them hides a lack of clarity on what the value really is. We should be able to trust the types.  

If we do not need the optional access operators, then they are noise and mask incorrect types. If we do need them, then our types are outdated, i.e. if we are sure `reason` is `IDataObject`, then it cannot be `undefined`, and if we are sure `routingError` is `AxiosError`, then it cannot be `undefined`. Does this keep failing without the optional access operators?

**Review by ivov — APPROVED** [review]
Thanks for fixing!

Btw not sure if this should be `(no-changelog)`

**Comment by Joffcom** [comment]
Got released with [`n8n@1.24.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.24.0)

**Pull request merged**